libraries/AP_HAL_Linux/RCInput.h

```diff
@@ -42,13 +44,13 @@ class RCInput : public AP_HAL::RCInput {
 
     // add some srxl input bytes, for RCInput over a serial port
     bool add_srxl_input(const uint8_t *bytes, size_t nbytes);
-    
+
 protected:
     void _process_rc_pulse(uint16_t width_s0, uint16_t width_s1);
     void _update_periods(uint16_t *periods, uint8_t len);
 
-    volatile uint32_t rc_input_count;
-    uint32_t last_rc_input_count;
+    std::atomic<unsigned int> rc_input_count;
+    std::atomic<unsigned int> last_rc_input_count;
```

Comment:
Any reason not to use *std::atomic_uint32_t*? Is it unsupported by our compiler?